pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: mountOptions is the mount time options
+	// including readonly
+	MountOpt string `json:"mountOptions,omitempty"`
+
+	// Optional: helper is the helper utility to mount command
+	// it can encapsulate the mount command,
+	// so mount can be run in a different namespace, if necessary
+	Helper string `json:"helper,omitempty"`
```

Comment:
Suggest leaving this out of first version.  We may be able to come up with more general mechanism.